src/lib/containers/entity_finder/EntityFinder.tsx

```diff
@@ -172,50 +111,80 @@ export const EntityFinder: React.FunctionComponent<EntityFinderProps> = ({
     type: EntityDetailsListDataConfigurationType.PROMPT,
   })
 
-  const isSelected = (entity: Reference) => {
-    return selectedEntities.some(
+  // If a type is selectable, it should be visible in the tree/list (depending on treeOnly)
+  const selectableAndVisibleTypesInTree = useMemo(
+    () => [...visibleTypesInTree, ...selectableTypes],
+    [visibleTypesInTree, selectableTypes],
+  )
+  const selectableAndVisibleTypesInList = useMemo(
+    () => [...visibleTypesInList, ...selectableTypes],
+    [visibleTypesInList, selectableTypes],
+  )
```

Comment:
If a type is selectable, then it should always be visible. This reduces the risk of error when configuring this component, where I've found that it's really easy to include an entity type as "selectable" but omit it from "visible"